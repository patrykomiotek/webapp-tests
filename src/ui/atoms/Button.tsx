import { ComponentPropsWithoutRef } from 'react';

type Props = ComponentPropsWithoutRef<'button'> & {};

export const Button = ({ children, ...rest }: Props) => {
  return (
    <button
      className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
      {...rest}
    >
      {children}
    </button>
  );
};
