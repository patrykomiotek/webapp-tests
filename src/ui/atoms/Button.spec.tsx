import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import { Button } from './Button';

describe('Button component', () => {
  it('should match snapshot', () => {
    const { container } = render(<Button>Click me</Button>);

    expect(container).toMatchSnapshot();
  });

  it('should pass accessibility check', async () => {
    const { container } = render(<Button>Click me</Button>);

    const result = await axe(container);

    expect(result).toHaveNoViolations();
  });
});
