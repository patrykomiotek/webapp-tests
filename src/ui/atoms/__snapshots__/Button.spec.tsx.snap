// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > should match snapshot 1`] = `
<div>
  <button
    class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
  >
    Click me
  </button>
</div>
`;
